### ISSUE TITLE
LessSpecificImplementedReturnType false-positive

### ISSUE THREAD

**Issue by fluffycondor** [weirdan/doctrine-psalm-plugin#71]
LessSpecificImplementedReturnType - src/Repository/SomethingRepository.php:9:12 - The inherited return type 'list<App\Entity\Something>' for Doctrine\ORM\EntityRepository::findAll is more specific than the implemented return type for Doctrine\ORM\EntityRepository::findall 'array<array-key, App\Entity\Something>'

Code in my SomethingRepository:
```php
/**
 * @method Something[]    findAll() (this PHPDoc was auto-generated by Symfony)
 * @extends ServiceEntityRepository<Something>
 */
class SomethingRepository extends ServiceEntityRepository
{
 */
```

This is a snippet from Doctrine\ORM\EntityRepository
```php
/**
 * Finds all entities in the repository.
 *
 * @return array The entities.
 */
public function findAll()
{
    return $this->findBy([]);
}
```
Just an array.

This is from Doctrine\Persistence\ObjectRepository that implemented by Doctrine\ORM\EntityRepository:
```php
/**
 * Contract for a Doctrine persistence layer ObjectRepository class to implement.
 *
 * @template T
 */
interface ObjectRepository
{
.....

    /**
     * Finds all objects in the repository.
     *
     * @return array<int, object> The objects.
     *
     * @psalm-return T[]
     */
    public function findAll();
....
}
```
So it returns just T[], Something[] in my case.

Looks like there is a problem in the plugin's stub files.

**Comment by weirdan** [comment]
As far as I can see (based on comments in [this `doctrine/persistence` PR](https://github.com/doctrine/persistence/pull/104)), there's indeed a problem with `ObjectRepository` stub where `findAll()` should return `T[]`, but there's no issue with descendant return types. ORM repository actually returns `list<T>`.

**Comment by fluffycondor** [comment]
Is there any info when there will be a new release with all of the bugfixes? :)

**Comment by weirdan** [comment]
I want to fix remaining deprecation errors first, so it would be some time next week.

**Comment by nsetyo** [comment]
@weirdan I'm still getting this error
![image](https://user-images.githubusercontent.com/4555074/151728724-9a8916ed-5df6-4a3e-8dde-8f2fb04b4e4f.png)

Based what I see `ServiceEntityRepository` extends `EntityRepository` and in `EntityRepository` findAll still return `list<T>`
How should I fix this?
